experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -184,8 +184,10 @@ export function createObjectProxy<TSchema extends ObjectNodeSchema, TTypes exten
 					return false;
 				}
 
-				// TODO: Is it safe to assume 'content' is a LazyEntity?
-				const field = getBoxedField(content as LazyEntity, key as FieldKey, fieldSchema);
+				// TODO: Is it safe to assume 'content' is a LazyObjectNode?
+				assert(content instanceof LazyObjectNode, "invalid content");
+				assert(typeof key === "string", "invalid key");
+				const field = getBoxedField(content, brand(key), fieldSchema);
```

Comment:
prefer using brand to conver to branded types over as, since as will accept many types that are not valid, and thus is less safe than brand.

Note that in this case, symbols are legal here, and are not valid as field keys: using brand discovered this, and thus an assert was added above.